File: api/models/spec.py
from pydantic import BaseModel, Field
from typing import List, Dict, Optional

class Behavior(BaseModel):
    validation: List[str] = []
    actions: List[str] = []
    api_calls: List[str] = []
    responses: Dict[str, str] = {}

class ArchitectSpec(BaseModel):
    goal: str = ""
    type_hypothesis: List[str] = []
    inputs: List[str] = []
    outputs: List[str] = []
    ui: List[str] = []
    state: List[str] = []
    behavior: Behavior = Field(default_factory=Behavior)

class MapRequest(BaseModel):
    user_text: str
    current_field: str
    current_spec: ArchitectSpec

class MapResponse(BaseModel):
    updated_spec: ArchitectSpec
    next_field: str
    next_question: str
    suggestions: List[str]
    is_complete: bool

File: src/types/spec.ts
export type SpecNode = 'goal' | 'type' | 'inputs' | 'ui' | 'actions' | 'state' | 'validation' | 'api' | 'responses' | 'outputs';

export interface Behavior {
  validation: string[];
  actions: string[];
  api_calls: string[];
  responses: Record<string, string>;
}

export interface ArchitectSpec {
  goal: string;
  type_hypothesis: string[];
  inputs: string[];
  outputs: string[];
  ui: string[];
  state: string[];
  behavior: Behavior;
}

export const PRIORITY_ORDER: SpecNode[] = [
  'goal',
  'type',
  'inputs',
  'ui',
  'actions',
  'state',
  'validation',
  'api',
  'responses',
  'outputs'
];

export const NODE_LABELS: Record<SpecNode, string> = {
  goal: 'Goal',
  type: 'Type',
  inputs: 'Inputs',
  ui: 'User Interface',
  actions: 'Actions',
  state: 'System State',
  validation: 'Validation',
  api: 'API Calls',
  responses: 'Responses',
  outputs: 'Outputs'
};
